codegen/src/main/twirl/templates/ScalaClient/Client.scala.txt

```diff
@@ -7,27 +7,37 @@
 @akka.grpc.gen.Constants.DoNotEditComment
 package @service.packageName
 
-import scala.concurrent.{ ExecutionContext, Promise }
+import scala.concurrent.ExecutionContext
 
 import io.grpc.ManagedChannel
 import io.grpc.MethodDescriptor
 
 import akka.grpc.GrpcClientSettings
 
 import akka.grpc.scaladsl.AkkaGrpcClient
-import akka.grpc.scaladsl.SingleResponseRequestBuilder
-import akka.grpc.scaladsl.StreamResponseRequestBuilder
 
-import akka.grpc.internal.ScalaUnaryRequestBuilder
-import akka.grpc.internal.ScalaBidirectionalStreamingRequestBuilder
-import akka.grpc.internal.ScalaClientStreamingRequestBuilder
-import akka.grpc.internal.ScalaServerStreamingRequestBuilder
 import akka.grpc.internal.Marshaller
 import akka.grpc.internal.NettyClientUtils
 import akka.grpc.internal.ClientState
 
 import akka.stream.Materializer
 
+@{
+def withSingleResponse(stmt: String) = Set("import akka.grpc.scaladsl.SingleResponseRequestBuilder", stmt)
+def withStreamResponse(stmt: String) = Set("import akka.grpc.scaladsl.StreamResponseRequestBuilder", stmt)
+service.methods.map(_.methodType).foldLeft(Set.empty[String]) { (imports, method) =>
```

Comment:
Rather than a full `fold` I think a `flatMap` might be sufficient here and a bit simpler?

Reply:
Sure!